homeassistant/components/lifx/manager.py

```diff
@@ -345,8 +349,21 @@ async def start_effect(
         elif service == SERVICE_EFFECT_COLORLOOP:
 
             brightness = None
+            saturation_max = None
+            saturation_min = None
+
             if ATTR_BRIGHTNESS in kwargs:
                 brightness = convert_8_to_16(kwargs[ATTR_BRIGHTNESS])
+            elif ATTR_BRIGHTNESS_PCT in kwargs:
+                brightness = convert_8_to_16(
+                    round(255 * kwargs[ATTR_BRIGHTNESS_PCT] / 100)
```

Comment:
Should this also use `int` for symmetry with the saturation values (or the other way around, have `saturation` use round)?

Reply:
You'd think, but I'm using the same methods used by the existing `find_hsbk` method in `util.py`: https://github.com/home-assistant/core/blob/dev/homeassistant/components/lifx/util.py#L104-L125 so this actually maintains consistency with the existing conversions done elsewhere.

Reply:
And as much as I really want to change all of those calculations to be more accurate with how LIFX actually stores color data, doing so would change the user experience and configuration and I'm extremely reticent to do that.

Reply:
Thanks for the clarification! In the spirit of DRY/SSOT, should that find_hsbk computation be split into functions that would be reused e.g. here?

Reply:
Perhaps, but that would be a significant refactoring in and of itself that it would be better done in a dedicated PR, as there is way more than just this that could benefit from that sort of deduplication of effort.

Reply:
If that's something you want to do, please do it in another PR to avoid making this one bigger